### DISCUSSION TITLE
Intellisense Errors in VS2022

### DISCUSSION THREAD

**Discussion by nsrahmad** [statiqdev/Discussions#186] [Q&A] [answered]
Hi,
I am new to C# and Visual Studio. I am setting up Statiq to use for my blog. After following the official tutorial and installing the CleanBlog theme, everything is running fine. But in visual studio 2022, when I open .cshtml file I see errors like:
![Screenshot 2023-01-05 210118](https://user-images.githubusercontent.com/10996250/210820530-f650f350-6abb-40a0-8d3c-f151d0eea85a.png)
Is there a way to get rid of these errors?

**Comment by daveaglick** [comment, marked as answer]
You're in luck! For background, the problem is that the Visual Studio Razor engine builds Razor pages in the background and makes a bunch of assumptions about what sort of base class is being used for the Razor pages, what namespaces are available, etc. This works for like 99% of the typical Razor uses via ASP.NET. Obviously Statiq isn't ASP.NET, and on top of that, Statiq does some customization of that Razor process that the built-in Visual Studio design-time Razor engine has no idea about (such as changing the Razor view base class).

Thankfully there actually is a way to tell the Visual Studio Razor Intellisense engine about the correct namespace imports and base Razor class that Statiq uses. The instructions are here: https://www.statiq.dev/guide/content-and-data/template-languages/razor#imports-file-and-intellisense. The important bit is:

> ...create a `_ViewImports.cshtml` file in your input folder:

```cshtml
@using Statiq.Common
@using Statiq.Razor
@using Statiq.Web
@using Statiq.Web.Pipelines
@using Microsoft.Extensions.Logging

@inherits StatiqRazorPage<IDocument>
```

That should get you going, let me know if you have any trouble after this. Note that you may need to tweak the `_ViewImports.cshtml` file with your own `@using` statements depending on what you're doing in your Statiq Razor files.

**Reply by nsrahmad** [reply]
Thank you. This makes perfect sense. Everything is working fine for now 🙏 .
